File: verifier-cli/src/main/java/com/jetbrains/pluginverifier/verifiers/instruction/InstructionVerifier.java
package com.jetbrains.pluginverifier.verifiers.instruction;

import com.intellij.structure.resolvers.Resolver;
import com.jetbrains.pluginverifier.verifiers.VContext;
import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.AbstractInsnNode;
import org.objectweb.asm.tree.ClassNode;
import org.objectweb.asm.tree.MethodNode;

/**
 * @author Dennis.Ushakov
 */
public interface InstructionVerifier extends Opcodes {
  void verify(ClassNode clazz, MethodNode method, AbstractInsnNode instr, Resolver resolver, VContext ctx);
}

File: verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/instruction/InstructionVerifier.kt
package com.jetbrains.pluginverifier.verifiers.instruction

import com.jetbrains.pluginverifier.verifiers.VerificationContext
import com.jetbrains.pluginverifier.verifiers.resolution.Method
import org.objectweb.asm.tree.AbstractInsnNode

interface InstructionVerifier {
  fun verify(method: Method, instructionNode: AbstractInsnNode, context: VerificationContext)
}